src/io/flutter/preview/PreviewView.java

```diff
@@ -297,78 +312,22 @@ public void invokePopup(Component comp, int x, int y) {
           }
         }
 
-        // The corresponding tree item has just been selected.
-        // Wait short time for receiving assists from the server.
-        for (int i = 0; i < 20 && actionToChangeMap.isEmpty(); i++) {
-          Uninterruptibles.sleepUninterruptibly(5, TimeUnit.MILLISECONDS);
-        }
-
-        final DefaultActionGroup group = new DefaultActionGroup();
-        boolean hasAction = false;
-        if (actionCenter.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionCenter));
-        }
-        if (actionPadding.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionPadding));
-        }
-        if (actionColumn.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionColumn));
-        }
-        if (actionRow.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionRow));
-        }
-        if (actionContainer.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionContainer));
-        }
-        group.addSeparator();
-        if (actionExtractMethod.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionExtractMethod));
-        }
-        if (actionExtractWidget.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionExtractWidget));
-        }
-        group.addSeparator();
-        if (actionMoveUp.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionMoveUp));
-        }
-        if (actionMoveDown.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionMoveDown));
-        }
-        group.addSeparator();
-        if (actionRemove.isEnabled()) {
-          hasAction = true;
-          group.add(new TextOnlyActionWrapper(actionRemove));
-        }
-
-        // Don't show the empty popup.
-        if (!hasAction) {
-          return;
-        }
-
-        final ActionManager actionManager = ActionManager.getInstance();
-        final ActionPopupMenu popupMenu = actionManager.createActionPopupMenu(ActionPlaces.UNKNOWN, group);
-        popupMenu.getComponent().show(comp, x, y);
+        widgetEditToolbar.createPopupMenu(comp, x, y);
```

Comment:
The popup menu is displayed identically as before, this code has just moved to the toolbar class.